tests/test_spatial_tools.py

```diff
@@ -178,6 +208,136 @@ def make_gdal_hillshade(filepath) -> np.ndarray:
     # Make sure that this doesn't create weird division warnings.
     xdem.spatial_tools.hillshade(dem.data, dem.res)
 
+class TestRobustFitting:
+
+    @pytest.mark.parametrize("pkg_estimator", [('sklearn','Linear'), ('scipy','Linear'), ('sklearn','Theil-Sen'),
+                                           ('sklearn','RANSAC'),('sklearn','Huber')])
+    def test_robust_polynomial_fit(self, pkg_estimator: str) -> None:
+
+        np.random.seed(42)
+
+        # x vector
+        x = np.linspace(1, 10, 1000)
+        # exact polynomial
+        true_coefs = [-100, 5, 3, 2]
+        y = true_coefs[0] + true_coefs[1] * x + true_coefs[2] * x ** 2 + true_coefs[3] * x ** 3
+
+        coefs, deg = xdem.spatial_tools.robust_polynomial_fit(x, y, linear_pkg=pkg_estimator[0], estimator=pkg_estimator[1], random_state=42)
+
+        assert deg == 3 or deg == 4
+        assert np.abs(coefs[0] - true_coefs[0]) <= 100
+        assert np.abs(coefs[1] - true_coefs[1]) < 5
+        assert np.abs(coefs[2] - true_coefs[2]) < 2
+        assert np.abs(coefs[3] - true_coefs[3]) < 1
+
+    def test_robust_polynomial_fit_noise_and_outliers(self):
+
+        np.random.seed(42)
+
+        # x vector
+        x = np.linspace(1,10,1000)
+        # exact polynomial
+        true_coefs = [-100, 5, 3, 2]
+        y = true_coefs[0] + true_coefs[1] * x + true_coefs[2] * x**2 + true_coefs[3] * x**3
+        # add some noise on top
+        y += np.random.normal(loc=0,scale=3,size=1000)
+        # and some outliers
+        y[50:75] = 0
+        y[900:925] = 1000
+
+        # test linear estimators
+        coefs, deg = xdem.spatial_tools.robust_polynomial_fit(x,y, estimator='Linear', linear_pkg='scipy', loss='soft_l1', f_scale=0.5)
+
+        # scipy solution should be quite robust to outliers/noise (with the soft_l1 method and f_scale parameter)
+        # however, it is subject to random processes inside the scipy function (couldn't find how to fix those...)
+        assert deg == 3 or deg == 4 # can find degree 3, or 4 with coefficient close to 0
+        assert np.abs(coefs[0] - true_coefs[0]) < 3
+        assert np.abs(coefs[1] - true_coefs[1]) < 3
+        assert np.abs(coefs[2] - true_coefs[2]) < 1
+        assert np.abs(coefs[3] - true_coefs[3]) < 1
+
+        # the sklearn Linear solution with MSE cost function will not be robust
+        coefs2, deg2 = xdem.spatial_tools.robust_polynomial_fit(x,y, estimator='Linear', linear_pkg='sklearn', cost_func=mean_squared_error, margin_improvement=50)
+        assert deg2 != 3
+        # using the median absolute error should improve the fit, but the parameters will still be hard to constrain
+        coefs3, deg3 = xdem.spatial_tools.robust_polynomial_fit(x,y, estimator='Linear', linear_pkg='sklearn', cost_func=median_absolute_error, margin_improvement=50)
+        assert deg3 == 3
+        assert np.abs(coefs3[0] - true_coefs[0]) > 50
```

Comment:
It's a bit weird to test that a method will be "at least as bad as..." !
Or did you mean the comparison sign to be the other way around?

Reply:
Well adding outliers mean the non-robust, "classic" estimators don't perform well. So yes, this is actually what I check :sweat_smile: 
I added some comments to make the meaning of the comparison clearer